schema_interface_test.go

```diff
@@ -0,0 +1,129 @@
+package kindsys_test
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"cuelang.org/go/cue"
+	"cuelang.org/go/cue/cuecontext"
+	"github.com/grafana/kindsys"
+	"github.com/grafana/thema"
+	"github.com/rogpeppe/go-internal/txtar"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestValidSchemas(t *testing.T) {
```

Comment:
Comment for the future (just an idea, but I'm happy to merge it as-is now): I guess we'll need a mechanism to bulk update all the golden files when needed

Reply:
one we copy over the txtar framework and start using that, I think we should be good